src/extension-support/extension-manager.js

```diff
@@ -108,6 +112,27 @@ class ExtensionManager {
         return this._loadedExtensions.has(extensionID);
     }
 
+    /**
+     * Synchronously load an internal extension (core or non-core) by ID. This call will
+     * fail if the provided id is not does not match an internal extension.
+     * @param {string} extensionId - the ID of an internal extension
+     */
+    loadExtensionIdSync (extensionId) {
+        if (builtinExtensions.hasOwnProperty(extensionId)) {
+            /** @TODO dupe handling for non-builtin extensions. See commit 670e51d33580e8a2e852b3b038bb3afc282f81b9 */
+            if (this.isExtensionLoaded(extensionId)) {
+                const message = `Rejecting attempt to load a second extension with ID ${extensionId}`;
+                log.warn(message);
+                return; // TODO Do we want to throw an error here?
```

Comment:
Good question... my gut feeling is that trying to load a second copy of an extension isn't a "fatal" error since the goal of loading an extension is to gain access to the blocks, and after this path is done the blocks should be available in the dupe case because they already were available before the call. I'm open to further discussion though :)

Reply:
I think that sounds reasonable!

```suggestion
                return;
```

Reply:
Should we make it so that the analogous code in loadExtensionURL doesn't reject the promise here?

Reply:
Oh. Hmm... I always find it awkward when I disagree with my past self... ;)

Yeah, I guess I would recommend just returning an empty `Promise.resolve()` from `loadExtensionURL` in the equivalent branch. I guess maybe I made that reject before I realized that rejecting a promise is just as "serious" as throwing an exception.

Reply:
I think this plan makes sense.

Reply:
I don't think the change in `loadExtensionURL` needs to block this PR; we can improve consistency in a later change if you'd like. I have this vague sense that we'll be actively working in this area for a while ;)